### PR TITLE
Menu Page: contains various components

### PR DESCRIPTION
**Pull request by Muniba-Ahmed**
New branch to merge in

### CODE REVIEW

**Comment by Muniba-Ahmed** [comment]
Screen Shot 2023-06-06 at 10 23 49 AM

media queries, changed font size, mapped price based on type of item.

**Comment by Muniba-Ahmed** [comment]
Screen Shot 2023-06-07 at 2 32 40 PM
 
Modal with name, image, description, price, & inventory count

**Comment by AdnanCodes** [comment]
Modal is looking great though the menu page layout is too spread out 
image

Perhaps put the items into another container so it makes look tighter in bigger screens

File: src/components/GridItem.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+              </CardContent>
+            </CardActionArea>
+          </Card>
+        </Grid>
+      ))}
+
+      <Modal open={selectedItem !== null} onClose={handleCloseModal}>
+        <Box sx={style}>
+          <div>
+            <Typography variant="h5">{selectedItem?.name}</Typography>
```
Comment (AdnanCodes) [inline src/components/GridItem.js:70]: is there need of optional chaining `?`

File: src/helpers/ItemList.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+      break;
+  }
+  return {
+    ...obj,
+    price,
+    inventory: 10,
+  };
+});
+
+console.log("is this working", itemList2);
```
Comment (AdnanCodes) [inline src/helpers/ItemList.js:202]: If it this working, let's avoid keeping any console.logs

Comment (Muniba-Ahmed) [reply src/helpers/ItemList.js:202]: console.log removed.

File: src/components/GridItem.js (hunk 1 above)
Comment (AdnanCodes) [reply src/components/GridItem.js:70]: Good, you need this if the data coming thru is async; sometimes it is best to do it at a bit higher level so you don't need that many optional chaining for singular data object

File: src/components/GridItem.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    setSelectedItem(item);
+  };
+
+  const handleCloseModal = () => {
+    setSelectedItem(null);
+  };
+
+  return (
+    <Grid container rowSpacing={3} columnSpacing={{ xs: 1, sm: 2, md: 3 }}>
+      {itemList2.map((item, index) => (
```
[the remark this answers is not in the document]
Comment (Muniba-Ahmed) [reply src/components/GridItem.js:40]: itemList2 was the new array of objects after i added the price & inventory key/values for all the baked goods. i have renamed it to itemDetailsList. I have also renamed the items to menuItems

**Comment by Muniba-Ahmed** [comment]
Screen Shot 2023-06-09 at 4 01 27 PM

implemented filtered tabs. Still struggling with spacing/design.

File: src/components/GridItem.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  p: 4,
+};
+
+const menuContainerStyle = {
+  padding: 5,
+  margin: 5,
+};
+
+function GridItem({ itemDetailsList }) {
+  const [selectedItem, setSelectedItem] = useState(null);
```
Comment (AdnanCodes) [inline src/components/GridItem.js:33]: Lets destructure selectedItem to this 

`const { name, description, image, price } = selectedItem`

Apply this `menuItem` as well

File: src/components/Modal.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
```
Comment (AdnanCodes) [inline src/components/Modal.js:1]: Is this file being used anywhere?

File: src/helpers/ItemDetailsList.js [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+      break;
+    case "icedCoffee":
+      price = 6;
+      break;
+    default:
+      price = 1;
+      break;
+  }
+  return {
+    ...obj,
```
Comment (AdnanCodes) [inline src/helpers/ItemDetailsList.js:196]: Good use of spread operator nicely done! 🙌🏽

File: src/pages/Menu.js [hunk 7]
```diff
@@ -1,7 +1,117 @@
-import React from "react";
+import React, { useState } from "react";
```
Comment (AdnanCodes) [inline src/pages/Menu.js:1]: may be this file should use name of MenuTabs; setup is quite style intensive

**Review by AdnanCodes — CHANGES_REQUESTED** [review]
Let me know what you think and let's chat before merging

File: src/pages/Menu.js (hunk 7 above)
Comment (Muniba-Ahmed) [reply src/pages/Menu.js:1]: Agreed. I've changed it to MenuTabs

File: src/helpers/ItemDetailsList.js (hunk 6 above)
Comment (Muniba-Ahmed) [reply src/helpers/ItemDetailsList.js:196]: Thankyou!

File: src/components/Modal.js (hunk 5 above)
Comment (Muniba-Ahmed) [reply src/components/Modal.js:1]: I believe I removed it locally but didnt push those changes yet. The code for the modal got moved to "GridItem.js"

File: src/components/GridItem.js (hunk 4 above)
Comment (Muniba-Ahmed) [reply src/components/GridItem.js:33]: Destructured both selectedItem and menuItem.

**Review by AdnanCodes — APPROVED** [review]
Well done

**Pull request merged**